src/smtencoding/FStar.SMTEncoding.EncodeTerm.fst

```diff
@@ -1457,7 +1457,7 @@ and encode_formula (phi:typ) (env:env_t) : (term * decls_t)  = (* expects phi to
           encode_formula (U.unmeta phi) env
 
         | Tm_match(e, _, pats, _) ->
-           let t, decls = encode_match e pats mkFalse env encode_formula in
+           let t, decls = encode_match e pats mkUnreachable env encode_formula in
```

Comment:
When encoding a `match` to SMT, we encode it as an ITE with a default else case asserting False. This else case is not reachable, since other parts of the VC have already proved that the match is exhaustive. So, rather than encoding it as False, we now encoded it as `Unreachable` (which is macro for False, so semantically equivalent), which enables us to avoid putting a spurious label on it for the error report.